### DISCUSSION TITLE
BlitzWolf SHP6 15A last version KO

### DISCUSSION THREAD

**Discussion by saviothecnic** [ct-Open-Source/tuya-convert#1029] [General]
Hi I have always used tuya-convert with great happiness

To upgrade BlitzWolf SHP6 15A sockets with tasmota

But the latest orders I try from two different suppliers

While being the same make and model
they don't update me :(

How come what's happening
I have to wait for a new version of tuya-convert
in order to update the latest versions of tuya-convert
I currently have tuya-convert 2.4.5
I also tried the NAS-WR01W
and these don't update either :(
Thank you

**Comment by MacSass** [comment]
Hi,
there can be various reasons - one could be that the firmware on the tuya models is too new and can not be updates currently.
Also there are some other issues possible - at a minimum you would have to provide output or logs of your tuya-convert.

**Comment by saviothecnic** [comment]
Thanks for the quick response
I am desperate I bought 4 outlets from 4 different stores and all of them
do not update me I am afraid that all the ones that sell now
they are up to date and i am stuck to install tasmota :(
Question Cannot upgrade the original firmware
with an older one to be able to use Tuya convert?

I'm not very experienced if you tell me how
I will be happy to attach the log you ask for

**Comment by MacSass** [comment]
No, you can´t change the Tuya FW to an older version in order to flash it ... if you could, then we could flash directly ...

How do you flash? Whatever you use - there should be output from the script that should give you an error message indicating what is going on. Also in the "scripts" folder you find various .log files that you could attach to look it.

If your FW is too new to flash the latest version of the script should be telling you this in your output, or you can identify it in the log files looking at smarthack-psk.log

**Comment by saviothecnic** [comment]
No, maybe I misspelled I'm Italian and I'm using google translate

I meant if it is possible on the socket to load the original firmware of the house but previous to then be able to show it to tuya convert I do not know if there is an archive or procedure to do this or if we just have to wait for a new version of tuya covnert to come out that also works with the latest versions

**Comment by MacSass** [comment]
The line 
ID: 0242416f68626d643661473931494652310b1b5a68297bd82482a6a2f4fc878d5cf17c9499bdd7c02bd2c9658284d65c0a
in you log indicates that the FW is too new ... you cannot update it currently.
See here: https://github.com/ct-Open-Source/tuya-convert/issues/483

If you read the output of the script you should see a message that your FW is too new ...

And again: There is not way to go back to an older firmware ... at least no easy way that would work over the air.

We will have to wait for a new tuya-convert or you need to flash by HW serial interface. 
See here for updates to this: https://github.com/ct-Open-Source/tuya-convert/wiki/Collaboration-document-for-PSK-Identity-02

**Comment by saviothecnic** [comment]
These sockets are too compact
I updated a few by removing and soldering the wires
But then something always gets ruined especially antenna
When you open it you take it apart and put it back together
Okay I'll keep these outlets in the drawer
hoping that a new version of tuya convert will arrive, compatible with these new firmware and version of sockets
But at the moment I have no information to understand before buying
if a socket has the new or old fw compatible for upgrade
with current tyua conver exact?
the thing is just luck from what I understand :(

**Comment by saviothecnic** [comment]
![immagine](https://user-images.githubusercontent.com/6697807/150861708-7044bb97-169a-4749-96e2-369ce7b2bd67.png)

For your info the left socket is updated while the right one does not 

and oddly enough the socket that updated also has a higher FW 
![immagine](https://user-images.githubusercontent.com/6697807/150861887-604bced7-100c-4764-a1b5-691aa83ac102.png)

This is FW of the right one that does not update 
![immagine](https://user-images.githubusercontent.com/6697807/150862076-e9b03260-0f64-478c-bbab-05feaead8a7c.png)
